components/dateinput/keyboard-typing.md

```diff
@@ -0,0 +1,165 @@
+| [`AutoCorrectParts`](#autocorrectparts) | `bool` <br /> (`true`) | Sets if the Date Input should **change the value of invalid segments automatically**. |
+| [`AutoSwitchKeys`](#autoswitchkeys) | `List<object>` | The custom keys, which will **move the focus to the next date format segment**. |
+| [`AutoSwitchParts`](#autoswitchparts) | `bool` <br /> (`true`) | Sets if the caret will **move automatically to the next date format segment** when the user completes the current one. |
+| [`TwoDigitYearMax`](#twodigityearmax) | `int` <br /> (`68`) | The maximum year value that is considered part of the **current century**. |
+
+
+## AllowCaretMode
+
+`AllowCaretMode` sets if the users should see a **blinking caret** inside the Date Input, when possible. It's `false` by default.
+
+* When `false`, one date format segment is fully selected (highlighted) and there is no visible caret.
+* When `true`, users will see the caret when a date segment is partially complete, or when [`AutoSwitchParts`](#autoswitchparts) is `false`.
+
+See the [example](#example) below.
+
+
+## AutoCorrectParts
+
+`AutoCorrectParts` sets if the Date Input should **change the value of invalid segments automatically**. It's `true` by default. For example, if the value is `January 31` and the user changes the month to `February`, the component will auto-correct the day to `28`.
+
+* When `true`, the component will not allow the user to input an invalid date when using the arrow keys to increment and decrement the segment values.
+* When `false`, the component will allow the user to enter an invalid date and will show its invalid styling to hint the user to auto-correct the value on their own.
+
+See the [example](#example) below.
+
+
+## AutoSwitchKeys
+
+`AutoSwitchKeys` lists the keyboard keys, which will **move the focus to the next date format segment**. By default, this list is empty.
+
+The left and right arrows always change the focused segment, regardless of this parameter value.
+
+To define `Tab` as an auto-switch key, use its ASCII code `9`. Note that this may be unexpected for some keyboard users.
+
+<div class="skip-repl"></div>
+
+````CSHTML
+<TelerikDateInput AutoSwitchKeys="@( new List<object>() { "/", 9 } )" />
+````
+
+See the [complete example](#example) below.
+
+
+## AutoSwitchParts
+
+`AutoSwitchParts` sets if the caret will **move automatically to the next date format segment** when the user completes the current one. It's `true` by default.
+
+* When `true`, the the caret will **not** move for days and months if the segment value is equal to `1` or `2`, because the value may not be complete yet. In this case, the user should press an arrow key or an "auto switch key" to advance.
+* When `false`, the user should always advance to the next segment manually with the arrow keys or any of the `AutoSwitchKeys`.
+
+See the [example](#example) below.
+
+
+## TwoDigitYearMax
+
```

Comment:
Maybe we should add a note that this setting applies to the input only and the users can select a year that exceeds the `TwoDigitYearMax` from the calendar(applies for pickers). If they want to restrict the selection as well, they need to specify `Max`.

Reply:
I think this can deceive developers and create confusion in users. Indeed, you can select any year via the Calendar, but if you then decide to change it via the keyboard, the century will change and the user will not know about this.

So I think we should advertise `TwoDigitYearMax` only in the restricted use case. Nevertheless, I enhanced the documentation, based on your comment.